src/unix/fsevents.c

```diff
@@ -265,32 +265,36 @@ static void uv__fsevents_event_cb(ConstFSEventStreamRef streamRef,
 
```

Comment:
Can I suggest a few changes around lines 256 to 272 for performance and legibility reasons? E.g.:
```c
  if (strncmp(path, handle->realpath, handle->realpath_len) != 0)
    continue;
```
To:
```c
  if (len < handle->realpath_len)
    continue;

  if (memcmp(path, handle->realpath, handle->realpath_len) != 0)
    continue;

  if (len == handle->realpath_len)
    goto accept;  // see below
```
And:
```c
  if (handle->realpath_len > 1 || *handle->realpath != '/') {
```
To:
```c
  if (handle->realpath_len == 0)
    continue;  // but this is probably a bug in the first place?

  if (handle->realpath_len == 1)
    goto accept;  // handle->realpath == "/"

  if (path[handle->realpath_len] != '/')
    continue;

accept:
  path += handle->realpath_len;
  len -= handle->realpath_len;

  if (*path == '\0') {
    // path == realname, scan backwards to find basename
  }
```

Reply:
That doesn't quite code for the same set of conditions. I've reordered some of the statements to try to make the code intent more apparent. Does that read better now?